server/src/com/thoughtworks/go/server/materials/ConfigMaterialUpdater.java

```diff
@@ -0,0 +1,135 @@
+                                 MaterialChecker materialChecker,
+                                 ConfigMaterialUpdateCompletedTopic configCompletedTopic,
+                                 MaterialUpdateCompletedTopic topic,
+                                 MaterialService materialService,
+                                 SubprocessExecutionContext subprocessExecutionContext)
+    {
+        this.repoConfigDataSource = repoConfigDataSource;
+        this.materialChecker = materialChecker;
+        this.materialRepository = materialRepository;
+        this.configCompleted = configCompletedTopic;
+        this.topic = topic;
+        this.materialService = materialService;
+        this.subprocessExecutionContext = subprocessExecutionContext;
+
+        this.configCompleted.addListener(this);
+    }
+
+    @Override
+    public void onMessage(MaterialUpdateCompletedMessage message) {
+        Material material = message.getMaterial();
+        //MDU is done using the checkout, it has done db update and stored latest changes
+        // but MUS is still waiting for material updated message on MaterialUpdateCompletedTopic
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug(format("[Config Material Update] Config material update completed for material %s. Starting parse process", material));
+        }
+        try {
+            if(message instanceof MaterialUpdateFailedMessage)
+            {
+                MaterialUpdateFailedMessage failure = (MaterialUpdateFailedMessage)message;
+                LOGGER.warn(String.format("[Config Material Update] Cannot update configuration part because material update has failed. Reason: ",
+                        failure.getReason()));
+            }
+            else {
+                File folder = materialRepository.folderFor(material);
+                MaterialRevisions latestModification = materialRepository.findLatestModification(material);
+                Revision revision = latestModification.firstModifiedMaterialRevision().getRevision();
+
+                MaterialRevision lastParseRevision = getMaterialRevisionAtLastParseAttempt(message);
+                if (lastParseRevision == null) {
+                    //never parsed
+                    UpdateConfigurationFromCheckout(folder, revision, material);
+                } else if (latestModification.findRevisionFor(material.config())
+                        .hasChangedSince(lastParseRevision)) {
+                    // revision has changed. the config files might have been updated
+                    UpdateConfigurationFromCheckout(folder, revision, material);
+                } else {
+                    // revision is the same as last time, no need to parse again
+                }
+            }
+        }
+        finally {
+            // always post the original message further
+            // this will remove material from inProgress in MUS
+            topic.post(message);
+        }
+    }
+
```

Comment:
I wanted to play nice and not exclude any types of materials just because I don't use them. The effort is few lines for each type.
But there are integration tests which require some significant effort in which I totally ignored them. We'll see if anyone complains. I guess it is better to provide some, perhaps working, support than just throwing exception.